src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosDataRetriever.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * @author sshorser
+ *
+ */
+public abstract class PharosDataRetriever extends FileRetriever
+{
+	protected static final String SKIP_AMOUNT_TOKEN = "##SKIPAMOUNT##";
+
+	protected static final String TOP_AMOUNT_TOKEN = "##TOPAMOUNT##";
+
+	private static final String APPLICATION_JSON = "application/json";
+
+	// Pharos requested that we use the DEV endpoint for large requests, since it's not used by
+	// as many users, and I guess they don't want us slowing down the main endpoint.
+	private static String pharosEndpoint = "https://ncatsidg-dev.appspot.com/graphql";
+//	private static String pharosEndpoint = "https://pharos-api.ncats.io/graphql";
+
+	// Some examples of testing Pharos' GraphQL endpoint.
+	//curl 'https://pharos-api.ncats.io/graphql' -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Connection: keep-alive' -H 'DNT: 1' -H 'Origin: https://pharos-api.ncats.io' --data-binary '{"query":"query {\n  targets(filter: {facets: [{facet:\"Data Source\" values:[\"UniProt\"]}]}) {\n    count\n    filter\n    {\n      term\n    \tfacets {\n        facet\n        values\n      }\n    }\n    facets {\n      facet\n      values {\n        name\n        value\n      }\n    }\n    targets(top:100 skip:2 ) {\n      name\n      synonyms {\n        name\n        value\n      }\n    }\n  }\n}"}' --compressed
+	// curl 'https://ncatsidg-dev.appspot.com/graphql' -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Connection: keep-alive' -H 'DNT: 1' -H 'Origin: https://ncatsidg-dev.appspot.com' --data-binary '{"query":"query {  targets {    targets(top:100) {      uniprot    }  }}\n"}' --compressed
+	private static int batchSize = 1000;
+
+	@Override
+	protected void doHttpDownload(Path path, HttpClientContext context) throws Exception, HttpHostConnectException, IOException
+	{
+		int numRequests = 0;
+		boolean moreRecords = true;
+		try(FileWriter fw = new FileWriter(this.destination))
+		{
+			while (moreRecords)
+			{
+				// TODO: In the future, make FileRetriever in release-common-lib more generic so that it deals with HttpEntityEnclosingRequestBase
+				// and let subclasses optionally inject their own request object (HttpGet or HttpPost).
+				// But there just isn't time before Release 77 for that sort of elegant solution, so yes, there *is* some duplication of effort in this
+				// doHttpDownload method and the method it overrode.
+				HttpPost post = new HttpPost(PharosDataRetriever.pharosEndpoint);
+
+
+				// We will need to keep making posts until we have them all. Requests will get 100 items at a time.
+				RequestConfig config = RequestConfig.copy(RequestConfig.DEFAULT)
+											.setConnectTimeout(1000 * (int)this.timeout.getSeconds())
+											.setSocketTimeout(1000 * (int)this.timeout.getSeconds())
+											.setConnectionRequestTimeout(1000 * (int)this.timeout.getSeconds())
```

Comment:
Maybe a temp variable for this so it can be reused: `int millisecondsTimeOut = 1000 * (int) this.timeout.getSeconds();`